### ISSUE TITLE
Tests fail in Julia <= 1.4

### ISSUE THREAD

**Issue by fingolfin** [ulthiel/Combinatorics.jl#29]
As you can see [here](https://github.com/ulthiel/JuLie.jl/runs/1813723533?check_suite_focus=true#step:6:60) the tests fail in Julia <= 1.4, presumably because some method that was introduced in Julia 1.5 is used:
```
Kostka polynomials: Error During Test at /home/runner/work/JuLie.jl/JuLie.jl/test/runtests.jl:849
  Test threw exception
  Expression: charge([2, 1, 1, 2, 3, 5, 4, 3, 4, 1, 1, 2, 2, 3]) == 8
  MethodError: no method matching splice!(::Array{Int64,1}, ::Array{Int64,1})
  Closest candidates are:
    splice!(::Array{T,1} where T, !Matched::Integer) at array.jl:1375
    splice!(::Array{T,1} where T, !Matched::Integer, !Matched::Any) at array.jl:1375
    splice!(::Array{T,1} where T, !Matched::UnitRange{#s69} where #s69<:Integer) at array.jl:1422
    ...
  Stacktrace:
   [1] charge(::Array{Int64,1}, ::Bool) at /home/runner/work/JuLie.jl/JuLie.jl/src/kostka_polynomials.jl:309
   [2] charge at /home/runner/work/JuLie.jl/JuLie.jl/src/kostka_polynomials.jl:288 [inlined]
   [3] macro expansion at /home/runner/work/JuLie.jl/JuLie.jl/test/runtests.jl:849 [inlined]
   [4] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Test/src/Test.jl:1113 [inlined]
   [5] top-level scope at /home/runner/work/JuLie.jl/JuLie.jl/test/runtests.jl:818
 ```

Either need to fix the test resp. the tested code, or switch this package to require Julia >= 1.5 (and drop the CI tests for using older Julia versions)

**Comment by ulthiel** [comment]
I'm progressive and would say we require Julia >= 1.5. But maybe @schto223 should decide because it's his code. What's with splice?

**Comment by ulthiel** [comment]
I think this should be possible to fix. The documentation for splice in 1.5 is [here](https://docs.julialang.org/en/v1.5/base/collections/#Base.splice!) and there's a remark below.

**Comment by fingolfin** [comment]
The comment basically means that if you want to stay compatible with Julia <= 1.4, you can't use `splice!` -- resp. you'll need to provide the missing `splice!` method. Of course that's not hard to do either way, but somebody has to do it (and it certainly is even easier to just drop support for Julia <= 1.4 -- you decide :-)

**Comment by ulthiel** [comment]
Yes. The question is whether we really need it. It's just in one place. @schto223 can say.

**Comment by schto223** [comment]
I've replaced splice! with two lines of code. It doesn't seem to affect the performance and solved the Issue for Julia<=1.4
I created a pull request: #32

**Comment by ulthiel** [comment]
Great, thanks!
